### DISCUSSION TITLE
Synology Permission Issues

### DISCUSSION THREAD

**Discussion by tvtechdir** [immich-app/immich#13371] [Q&A] [answered]
Hey all, fairly stupid question but I've been hitting a wall all day and can't figure it out. I'm running Immich on an old 2012 MBP and want to leave the media on my Synology NAS. I've got everything working on the top level of the photo share on the Synology which is fine but messy with the uploads and thumbs and generic library folders. I want to clean it up by at least putting Immich's main sub-folders in an "Immich" folder in the photo share. Whenever I tweak the .env I get permission errors in the docker logs. So I tried creating another share, but that one would not work even on the top level.

What works: xxx.xxx.xxx.xxx:/volume1/photo
What does not work: xxx.xxx.xxx.xxx/volume1/photo/Immich or xxx.xxx.xxx.xxx/volume1/ImmichLibrary

I have to be missing a simple step. Can anyone lend some knowledge?

**Comment by petersuba** [comment]
Have you tried mounting the remote share and using the local file system path in the immich config?

**Reply by tvtechdir** [reply]
Hi Peter. Thanks for responding. The short answer is yes.

I followed [this](https://acceptdefaults.com/2020/03/16/mounting-a-synology-share-in-ubuntu/#synology-share-setup) tutorial to get a temporary NFS share mounted. I can successfully run the touch command on it on the Ubuntu machine on any share or sub folder of any share. I would assume the Immich docker compose instance would also be able to do so. I run the following command to create the temporary shares whether they work or not: sudo mount -t nfs 10.0.1.8:/volume1/photo /nfs/NASPhotos/.

My docker-compose and .env files are attached. It seems like a permissions thing with Immich because it's immich_server that's throwing the following error:

immich_server            | [Nest] 7  - 10/11/2024, 11:00:15 AM   ERROR [Microservices:StorageService] Failed to read upload/encoded-video/.immich: Error: ENOENT: no such file or directory, open 'upload/encoded-video/.immich'

[Immich Docker Compose.txt](https://github.com/user-attachments/files/17344265/Immich.Docker.Compose.txt)
[immich dotENV.txt](https://github.com/user-attachments/files/17344267/immich.dotENV.txt)

**Reply by petersuba** [reply, marked as answer]
I can't initially see an issue but to be fair I am not an expert, although I am running it with 2 Synologies and sometimes an additional MacBook running the ML as needed and one of the Synologies running the web server. 

However, the last few releases added .immich file checks and the release notes have inferred that some situations may have meant issues at startup related to them. According to the release note #13135 for v1.117.0 you can add an ENV option (`IMMICH_IGNORE_MOUNT_CHECK_ERRORS=true`) to skip this check, maybe try to set that variable to see if the issue is with that check only or if there is something more...

**Reply by tvtechdir** [reply]
I will give that a shot when I get a moment. Thanks for the suggestion!

**Reply by petersuba** [reply]
BTW I think what may also work for your situation is mounting the share as is, but passing the subdirectory to immich in the .env file as the mapped root. So, mount as you do now i.e. mount 10.0.1.8:/volume1/photo /nfs/NASPhotos/ but then in the .env file use
`UPLOAD_LOCATION=/nfs/NASPhotos/immich`

**Reply by tvtechdir** [reply]
Thank you so much for your help. IMMICH_IGNORE_MOUNT_CHECK_ERRORS=true did the trick. I will mark that as the answer. I had done a version of passing the subdirectory to the .env file, but I think was keeping it one-to-one with mounting the share.

Ignoring a Mount Check seems kind of disconcerting. Is that simply a bug or are folks losing data?

**Reply by petersuba** [reply]
> 
> Ignoring a Mount Check seems kind of disconcerting. Is that simply a bug or are folks losing data?

I am not too sure about the details of this. In my view, as this file serves to ensure that a mounted folder does actually exist, and since you are using a mounted remote folder, making this functionality work should be a good choice. You may want to report separately your configuration that fails regarding the .immich check and hopefully someone more in tune with the details of how it functions may be able to help - potentially even updating Immich to make it work as it may even be an edge case that it does not handle well at the moment.
